File: Tecnicell.Server/Models/Entity/PhoneRepair.cs
using System;
using System.Collections.Generic;

namespace Tecnicell.Server.Models.Entity;

public partial class PhoneRepair
{
    public string Imei { get; set; } = null!;

    public string? Brand { get; set; }

    public string? Name { get; set; }

    public string? CustomerName { get; set; }

    public string? CustomerId { get; set; }

    public string? CustomerNumber { get; set; }

    public string? ImageCode { get; set; }

    public decimal? Price { get; set; }

    public virtual Brand? BrandNavigation { get; set; }

    public virtual Image? ImageCodeNavigation { get; set; }

    public virtual ICollection<PhoneRepairHistory> PhoneRepairHistories { get; set; } = new List<PhoneRepairHistory>();
}

File: tecnicell.client/src/app/Interfaces/business/PhoneRepair.ts
import { Branch } from "./Branch";
import { PhoneBrand } from "./PhoneBrand";
import { Sale } from "./Sale";

export interface PhoneRepair {
    imei:                 string;
    brand:                string;
    customerName:         string;
    customerID:           string;
    customerNumber:       string;
    price:                number;
    brandNavigation:      PhoneBrand;
    phoneRepairHistories: PhoneRepairHistory[];
}

export interface PhoneRepairHistory {
    imei:               string;
    date:               Date;
    actionHistory:      string;
    toBranch:           string;
    description:        string;
    saleCode:           string;
    saleCodeNavigation: Sale;
    toBranchNavigation: Branch;
}
